{
  if (isDebug) {
    logger.afterInterceptor(target,args,result);
  }
  prepareAfterTrace(target,args,result,throwable);
  final Trace trace=traceContext.currentTraceObject();
  if (trace == null) {
    return;
  }
  try {
    doInAfterTrace(trace,target,args,result,throwable);
  }
 catch (  Throwable th) {
    if (logger.isWarnEnabled()) {
      logger.warn("after error. Caused:{}",th.getMessage(),th);
    }
  }
 finally {
    trace.traceBlockEnd();
  }
}
